mock/py/mockbuild/podman.py

```diff
@@ -1,13 +1,46 @@
 # -*- coding: utf-8 -*-
 # vim: noai:ts=4:sw=4:expandtab
 
+import logging
 import subprocess
 from contextlib import contextmanager
 
 from mockbuild.trace_decorator import getLog, traceLog
 from mockbuild import util
 from mockbuild.exception import BootstrapError
 
+
+def podman_check_native_image_architecture(image, logger=None):
```

Comment:
This can be reused:
```
In [1]: from mockbuild.podman import podman_check_native_image_architecture

In [2]: podman_check_native_image_architecture("fedora:38")
Image architecture linux/amd64 doesn't match system arch linux/ppc64le
Out[2]: False
```